internals/daemon/daemon.go

```diff
@@ -71,6 +71,10 @@ type Options struct {
 	// ServiceOuput is an optional io.Writer for the service log output, if set, all services
 	// log output will be written to the writer.
 	ServiceOutput io.Writer
+
+	// OverlordExtension is an optional struct used to extend the capabilities
```

Comment:
Super nitty comment: I would refer to it as an interface, not struct. Also optional does not exactly explain how to not use it. Here are some thoughts for inspiration.
```
// OverlordExtension is an interface allowing the extension of the default Overlord. If no extension is needed, nil should be supplied.
```

Reply:
See the comments of the other fields in the Options struct: optional means "either something or nil" (pretty sure this is a widely used Golang convention). And the default behavior in case of "nil" is not to extend (I think this is obvious).

Reply:
OK thanks. Makes sense!